### ISSUE TITLE
no-named-as-default with faux-decorators

### ISSUE THREAD

**Issue by prashaantt** [benmosher/eslint-plugin-import#183]
First off, thanks a ton for making this awesome plugin!

I have the following code:

```js
// Container.js

import { connect } from 'react-redux';

export class Container extends React.Component {}

export default connect()(Container);
```

```js
// App.js

import Container from './Container'; // Using exported name 'Container' as identifier for default export. (import/no-named-as-default)
```

As you can see, I'm avoiding using a decorator in `Container.js`, which otherwise could be written in ES7 as:

```js
// Container.js with decorator

import { connect } from 'react-redux';

@connect()
export default class Container extends React.Component {}
```

The second should work fine with the `no-named-as-default` rule, but would it be possible to make your linter aware of the curried `Component` with the first `export default`? (I'm presently working around it by removing the `export` around the `class Container`, which suppresses this warning.)

**Comment by prashaantt** [comment]
Wait a minute, I now realise this doesn't make sense. I could solve my problem by importing the default export under another name if I insist on doing it this way, so it's your plugin just being its helpful best by warning me to avoid any confusions arising with name conflicts.
